cli/tests/test_commit_template.rs

```diff
@@ -230,6 +230,72 @@ fn test_log_builtin_templates() {
     "###);
 }
 
+#[test]
+fn test_log_builtin_templates_colored() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let render =
+        |template| test_env.jj_cmd_success(&repo_path, &["--color=always", "log", "-T", template]);
+
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &[
+            "--config-toml=user.email=''",
+            "--config-toml=user.name=''",
+            "new",
+        ],
+    );
+
+    insta::assert_snapshot!(render(r#"builtin_log_oneline"#), @r###"
+    @  [1m[38;5;13mr[38;5;8mlvkpnrz[39m [38;5;9m(no email set)[39m [38;5;14m2001-02-03 04:05:08.000 +07:00[39m [38;5;12md[38;5;8mc315397[39m [38;5;10m(empty)[39m [38;5;10m(no description set)[39m[0m
+    ◉  [1m[38;5;5mq[0m[38;5;8mpvuntsm[39m [38;5;3mtest.user[39m [38;5;6m2001-02-03 04:05:07.000 +07:00[39m [1m[38;5;4m2[0m[38;5;8m30dd059[39m [38;5;2m(empty)[39m [38;5;2m(no description set)[39m
+    ◉  [1m[38;5;5mz[0m[38;5;8mzzzzzzz[39m [38;5;2mroot[39m [1m[38;5;4m0[0m[38;5;8m0000000[39m
+    "###);
+
+    insta::assert_snapshot!(render(r#"builtin_log_compact"#), @r###"
+    @  [1m[38;5;13mr[38;5;8mlvkpnrz[39m [38;5;9m(no email set)[39m [38;5;14m2001-02-03 04:05:08.000 +07:00[39m [38;5;12md[38;5;8mc315397[39m[0m
+    │  [1m[38;5;10m(empty)[39m [38;5;10m(no description set)[39m[0m
+    ◉  [1m[38;5;5mq[0m[38;5;8mpvuntsm[39m [38;5;3mtest.user@example.com[39m [38;5;6m2001-02-03 04:05:07.000 +07:00[39m [1m[38;5;4m2[0m[38;5;8m30dd059[39m
+    │  [38;5;2m(empty)[39m [38;5;2m(no description set)[39m
+    ◉  [1m[38;5;5mz[0m[38;5;8mzzzzzzz[39m [38;5;2mroot[39m [1m[38;5;4m0[0m[38;5;8m0000000[39m
+    "###);
+
+    insta::assert_snapshot!(render(r#"builtin_log_comfortable"#), @r###"
+    @  [1m[38;5;13mr[38;5;8mlvkpnrz[39m [38;5;9m(no email set)[39m [38;5;14m2001-02-03 04:05:08.000 +07:00[39m [38;5;12md[38;5;8mc315397[39m[0m
+    │  [1m[38;5;10m(empty)[39m [38;5;10m(no description set)[39m[0m
+    │
+    ◉  [1m[38;5;5mq[0m[38;5;8mpvuntsm[39m [38;5;3mtest.user@example.com[39m [38;5;6m2001-02-03 04:05:07.000 +07:00[39m [1m[38;5;4m2[0m[38;5;8m30dd059[39m
+    │  [38;5;2m(empty)[39m [38;5;2m(no description set)[39m
+    │
+    ◉  [1m[38;5;5mz[0m[38;5;8mzzzzzzz[39m [38;5;2mroot[39m [1m[38;5;4m0[0m[38;5;8m0000000[39m
+    "###);
+
+    insta::assert_snapshot!(render(r#"builtin_log_detailed"#), @r###"
+    @  Commit ID: [38;5;4mdc31539712c7294d1d712cec63cef4504b94ca74[39m
+    │  Change ID: [38;5;5mrlvkpnrzqnoowoytxnquwvuryrwnrmlp[39m
+    │  Author: [38;5;1m(no name set)[39m <[38;5;1m(no email set)[39m> ([38;5;6m2001-02-03 04:05:08.000 +07:00[39m)
```

Comment:
Thinking out loud: I'm not a huge fan of how this looks; we could adjust the template to say something like

    (no name set, no email set)

in this case. But this is outside the scope of this PR.

Reply:
Maybe even shorten a little bit more to `(no name, email set)`.

Reply:
(Resolving for now to make the merge gates happy.)

Reply:
I'm not sure. I first wanted to reword that to `(no name or email set)`. But I think the better English style would be `(neither name nor email set)`. Compared to that, I might prefer `(no name set, no email set)`?

In any case, it's not very important, we can go with whatever or see what others think.

Reply:
I would read `(no name, email set)` as "no name, [but] email is set"